llvm/include/llvm/IR/Constants.h

```diff
@@ -912,6 +912,11 @@ class BlockAddress final : public Constant {
   /// block must be embedded into a function.
   static BlockAddress *get(BasicBlock *BB);
 
+  /// Return a BlockAddress for the specified basic block, which may not be
+  /// part of a function. The specified type must match the type of the function
+  /// the block will be inserted into.
+  static BlockAddress *get(Type *Ty, BasicBlock *BB);
```

Comment:
Should we also add a corresponding helper for SandboxIR? It is the only way to create a blockaddress if it has not been inserted.

Reply:
I'd say "no unless it's actually needed". We should avoid creating blockaddresses for non-inserted blocks if possible.